compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -68,19 +68,22 @@ fn emit_frag_parse_err(
     kind: AstFragmentKind,
 ) {
     // FIXME(davidtwco): avoid depending on the error message text
-    if parser.token == token::Eof && e.message[0].0.expect_str().ends_with(", found `<eof>`") {
-        if !e.span.is_dummy() {
-            // early end of macro arm (#52866)
-            e.replace_span_with(parser.sess.source_map().next_point(parser.token.span));
-        }
+    if parser.token == token::Eof
+        && let DiagnosticMessage::Str(message) = &e.message[0].0
```

Comment:
This line should be fixed after the corresponding diagnostics is ported.

Reply:
Will this test fail when that happens?

Reply:
No, it will not. However, when that happens, UI tests such as [`src/test/ui/macros/macro-in-expression-context-2.stderr`](https://github.com/rust-lang/rust/blob/365578445c6771bb74bfd341c8c48867164a9992/src/test/ui/macros/macro-in-expression-context-2.stderr) fail. So, we will notice that this line needs to be fixed.